src/editor/controls/GitHubRepositoryControl.tsx

```diff
@@ -0,0 +1,117 @@
+import {BaseControl, Button, TextControl,} from '@wordpress/components';
+import {useRef} from '@wordpress/element';
+import {__} from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import useSWR, { mutate } from "swr";
+
+interface GitHubRepositoryControlProps {
+    value: string;
+    onChange: (value: string) => void;
+    onCodeFetched: (data: any) => void;
+}
+
+const fetcher = async (url: string) => {
+    const response = await apiFetch({
+        path: `code-block-pro/v1/code?url=${encodeURIComponent(url)}`,
+        method: 'GET',
+    });
+
+    return response.code;
+};
```

Comment:
This can stay, but it might be nicer to extract the hook out to its own file with other hooks, so it's reusable anywhere like:

```js
import { useGitHubSync } from './hooks/useGitHubSync'
// ...

const { code, loading, error } = useGitHubSync(url)
```

Also, using the `const functionsName = () => ` as you have here should be consistent elsewhere too.